notebook/autobuild_basic.ipynb

```diff
@@ -1367,15 +1367,163 @@
     "new_builder.clear_all_agents()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "73eb522e",
+   "metadata": {},
+   "source": [
+    "## Build agents with different backbone models\n",
+    "We support building agents with different backbone models. To achieve this, we need to first add a `profile` field to describe model's capability in the config list. Then we can specify agents' candidate models by passing a list of model names to `AgentBuilder()`. The builder manager will choose the most suitable model for each agent based on the agent/model profile and the task description."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 2,
+   "id": "0281711a",
+   "metadata": {},
    "outputs": [],
-   "source": [],
+   "source": [
+    "import json\n",
+    "\n",
+    "config_list = autogen.config_list_from_json(config_file_or_env, filter_dict={\"model\": \"gpt-4\"})\n",
+    "\n",
+    "# Add a profile field to gpt-4 model.\n",
+    "config_list[0][\"profile\"] = \"A pre-trained model that is good at code programming.\"\n",
+    "\n",
+    "# Add a new model that is good at mathematical problem solving.\n",
+    "config_list.extend(\n",
+    "    [\n",
+    "        {\n",
+    "            # Choose your model name.\n",
+    "            \"model\": \"microsoft/phi-2\",\n",
+    "            \"api_key\": \"NotRequired\",\n",
+    "            # You need to provide a short description of your model here.\n",
+    "            \"profile\": \"A pre-trained model that is good at sovling mathematical reasoning problems.\",\n",
+    "        }\n",
```

Comment:
A description alone is not enough. Context length, cost, required GPU resources, and benchmark ranking are also necessary for model selection.

Reply:
Yes, but context length and required GPU resources are crucial for successfully solving a task. We should add some restrictions or guidance for users to adopt this feature correctly, like splitting these fields out.

Reply:
Yeah, it makes sense to me. How do you prefer to use these crucial informations? Like longer context model has a higher priority, and check the gpu's avaliablity in runtime?

Reply:
> longer context model has a higher priority, and check the gpu's availability in runtime

LGTM!

Reply:
@LinxinS97 Mind reviewing again?